pkg/graveler/ref/manager.go

```diff
@@ -644,3 +644,61 @@ func (m *Manager) DeleteExpiredImports(ctx context.Context, repository *graveler
 	}
 	return errs.ErrorOrNil()
 }
+
+// Pull Requests logic
+
+func (m *Manager) getPullWithPredicate(ctx context.Context, repository *graveler.RepositoryRecord, pullID graveler.PullRequestID) (*graveler.PullRequest, kv.Predicate, error) {
+	type pullWithPred struct {
+		*graveler.PullRequest
+		kv.Predicate
+	}
+	key := fmt.Sprintf("GetPullRequest:%s:%s", repository.RepositoryID, pullID)
+	result, err := m.batchExecutor.BatchFor(ctx, key, m.maxBatchDelay, batch.ExecuterFunc(func() (interface{}, error) {
+		pullKey := graveler.PullRequestPath(pullID)
+		data := graveler.PullRequestData{}
+		pred, err := kv.GetMsg(context.Background(), m.kvStore, graveler.RepoPartition(repository), []byte(pullKey), &data)
+		if err != nil {
+			return nil, err
+		}
+		return &pullWithPred{graveler.PullRequestFromProto(&data), pred}, nil
+	}))
+	if errors.Is(err, kv.ErrNotFound) {
+		err = graveler.ErrPullRequestNotFound
+	}
+	if err != nil {
+		return nil, nil, err
```

Comment:
I guess we also do it in other places, but note that this means that the manager might return `kv` errors (where I would expect it to return `graveler` errors).

Reply:
That's correct, but should be handled in higher layers